encoder.go

```diff
@@ -50,3 +60,47 @@ func (e *Encoder) writeLiteralFieldWithoutNameReference(f HeaderField) {
 	e.buf = appendVarInt(e.buf, 7, uint64(len(f.Value)))
 	e.buf = append(e.buf, []byte(f.Value)...)
 }
+
+// Encodes a header field whose name is present in one of the
+// tables. TBit is true if the idx refers to the static table.
+func (e *Encoder) writeLiteralFieldWithNameReference(
+	f *HeaderField, idx int, NBit, TBit bool) {
```

Comment:
You don’t need NBit and TBit here. They are always false and true, respectively.

Reply:
Ok I'll remove them for this PR.
The T bit is strictly related to the dynamic table but I'm not sure the N bit is. For example the N bit is present in the encoding of a Literal Field Line Without Name Reference even though this doesn't use the dynamic table at all.
The RFC says 
```
The following bit, 'N', indicates whether an intermediary is 
permitted to add this field line to the dynamic table on subsequent hops.When
the 'N' bit is set, the encoded field line MUST always be encoded
with a literal representation.  In particular, when a peer sends a
field line that it received represented as a literal field line with
the 'N' bit set, it MUST use a literal representation to forward this
field line.  This bit is intended for protecting field values that
are not to be put at risk by compressing them; see Section 7 for more
details.
```

So even if our implementation doesn't use the dynamic table we should offer the caller the possibility to set the N bit so that the value is not cached by a subsequent proxy down the line.
That being said I also think this can be done in a subsequent PR.